src/OSPSuite.SimModelNative/src/BooleanFormula.cpp

```diff
@@ -261,18 +287,17 @@ void BooleanFormula::UpdateIndicesOfReferencedVariables()
 
 double AndFormula::DE_Compute (const double * y, const double time, ScaleFactorUsageMode scaleFactorMode)
 {
-	assert (m_SecondOperandFormula != NULL); //first operand is always set
-	return  (m_FirstOperandFormula->DE_Compute(y, time, scaleFactorMode) && m_SecondOperandFormula->DE_Compute(y, time, scaleFactorMode));
+	return calculate_binaryOperation(y, time, scaleFactorMode, logical_and<>{});
```

Comment:
Relevant changes are in `XXX::DE_Compute()` functions and in `XXX::DE_Jacobian()` functions (where applies)
For all boolean functions I just call `calculate_binaryOperation()` or `calculate_unaryOperation()` passing the logical operation itself as the last argument (e.g. here `logical_and<>{}`)